server/redeemer.go

```diff
@@ -0,0 +1,303 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/big"
+	gonet "net"
+	"net/url"
+	"sync"
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/livepeer/go-livepeer/eth"
+	"github.com/livepeer/go-livepeer/net"
+	"github.com/livepeer/go-livepeer/pm"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+const rpcTimeout = 8 * time.Second
+
+type redeemer struct {
+	recipient   ethcommon.Address
+	subs        sync.Map
```

Comment:
I guess we could use `event.Subscription` here , but since this is third party should we really be relying on it too much?